torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -458,6 +463,40 @@ std::string stacksToStr(const std::vector<std::string>& stacks, const char* deli
 
 } // namespace
 
+void reportBackendEventToActiveKinetoProfiler(
+    const int64_t start_time_us,
+    const int64_t end_time_us,
+    const int64_t debug_handle,
+    const at::RecordScope scope,
+    const std::string& event_name,
+    const std::string& backend_name) {
+  auto state_ptr = getProfilerTLSState();
+  if (!state_ptr) {
+    return;
+  }
+  auto ctx_ptr = std::make_unique<KinetoObserverContext>();
```

Comment:
While this is fine for the moment, I think we can optimize this a lot - it seems unnecessary to create the KinetoObserverContext in this case. In fact we can probably remove it from the profiler altogether...

Reply:
I dont know if you meant for the backend event reporting only or generally in the profiler itself but I completely agree. I thought the same that we dont need the ObserverContext. My understanding was that the observer context was used by callback where enter callback construct one, push some info and return the context. Then record_function calls exit callback with the context which populates more information. For the purposes of the API here, it is not needed.